README.md

```diff
@@ -109,12 +109,13 @@ Note: This is only recommended for those that wish to make code changes to Nanos
 
 Nanos doesn't need too many dependencies on Linux.
 
-To build you need to install nasm && qemu:
+To build you need to install nasm, qemu and go (If you don't want to install go, choose a different target, see below):
```

Comment:
This doesn't tell what target doesn't need go, I would be a bit more explicit, something along the lines of "installing go is not necessary if you only want to build the kernel"

Reply:
Go is also mentioned as dependency for Mac, so this is consistent with what is already there. 

Go is not strictly needed for all targets, but it's needed for the default target: 

In ´´´Makefile´´´ the default target uses go:
```makefile
5: # runtime tests / ready-to-use targets
6: TARGET=		webg
```

If you follow the instructions in Readme
```
sudo apt-get install qemu-system-x86 nasm
make run
```
the build fails.

I think it's not a great experience likes this.

So should we mention that go is not needed for all targets (for both Mac and PC description)?

Or as an alternative, should we use another default target, that doesn't need go? And then mention go at the section where it is explained how to build specific targets? Maybe web or hw ?

Reply:
I would keep the default target as-is, I just think "choose a different target" in the readme file is too vague ("target" might be interpreted as meaning a Makefile target, which is a different thing than the TARGET variable used when invoking make).
I'm OK if for simplicity you just add Go to the dependencies for building on Linux; but if you want to also add a note to say that Go is not needed when building/running a non-default target, this note should be more precise, e.g. mention the TARGET variable.

Reply:
https://github.com/nanovms/nanos/pull/2071/commits/8970938f3271613e485463483066fb536bdd88c3
- added a note about go to the mac and linux build instructions
- added TARGET variable explanation & link to overview in example section 
- added an overview of examples in the test/runtime dir